sdk/metric/internal/reuse_slice.go

```diff
@@ -0,0 +1,22 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package internal // import "go.opentelemetry.io/otel/sdk/metric/internal"
+
+func ReuseSlice[T any](slice []T, n int) []T {
```

Comment:
Why not just include this in `sdk/metric/pipeline.go` as `reuseSlice`? It seems that's the only place this is used.

Reply:
Because it will be used again in every Aggregator to resize []DataPoint and []HistogramDataPoints (any buckets and bounds).